scrunch/datasets.py

```diff
@@ -1441,12 +1480,32 @@ class Variable(object):
 
     CATEGORICAL_TYPES = {'categorical', 'multiple_response', 'categorical_array'}
 
-    def __init__(self, resource, dataset_resource):
-        self.resource = resource
-        self.url = self.resource.self
-        self.dataset = Dataset(dataset_resource)
+    def __init__(self, var_tuple, dataset):
+        self.shoji_tuple = var_tuple
+        self.is_instance = False
+        self._resource = None
+        # pass an Entity class to save time
+        if isinstance(dataset, pycrunch.shoji.Entity):
+            self.dataset = dataset
+        else:
+            # Need to instantiate Dataset for ds.order method
+            self.dataset = Dataset(dataset)
```

Comment:
Why do we need the `Dataset` class here? Why can we not just proceed with the dataset shoji resource?

The Dataset class has many user facing methods that we don't need here, we should be able to proceed with direct shoji operations to the resource from the internal implementation of Variable here